File: public/dsa/two_pointers/unique_elements_in_sorted_array/naive/unique_elements_in_sorted_array.cpp

#include <bits/stdc++.h>
using namespace std;

int removeDuplicates(vector<int> &A) {
    unordered_set<int> duplicateRemoved;
    for (int i = 0; i < A.size(); i++) {
        duplicateRemoved.insert(A[i]);
    }
    return duplicateRemoved.size();
}

// tests-start
int main() {
    vector<int> A = {1, 2, 3, 3, 3, 4, 5, 5};
    int size = removeDuplicates(A);
    cout << size << endl;
    return 0;
}
// tests-end

/*output
5
*/

File: public/dsa/two_pointers/unique_elements_in_sorted_array/naive/unique_elements_in_sorted_array.py
from typing import List


class Solution:
    def removeDuplicates(self, A: List[int]) -> int:
        return len(set(A))


# tests-start
A = [1, 2, 3, 3, 3, 4, 5, 5]
sol = Solution()
print(sol.removeDuplicates(A))
# tests-end

'''output
5
'''
